### PR TITLE
network separation and integrated

### PR DESCRIPTION
**Pull request by zkn111**
Change-Id: I3010a82b9cc6238b0de55b4dc2094a4253877682

network separation and integrated
@PengJiyuan

### CODE REVIEW

**Comment by zkn111** [comment]
admin:
add network 、subnet 、router new function and paging and search 
all function on the basis of project ID

@PengJiyuan

File: modules/network/lang.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "en": {
+    "create_subnet": "Create Subnet",
+    "cidr_tip": "Please fill in the blank matching format such as 192.168.1.0/24",
+    "enable_security": "Enable security restrict(Once be set, it can not be changed)",
+    "enable_vlan": "Vlan network",
+    "external_network": "External Network",
```
Comment (PengJiyuan) [inline modules/network/lang.json]: the key external_network is duplicate in 2 files:
halo/client/applications/admin/modules/network-agents/lang.json
halo/client/applications/admin/modules/network/lang.json

File: locale/lang.json [hunk 2]
```diff
@@ -117,7 +119,7 @@
     "cancel": "取消",
     "compute_service": "计算服务",
     "confirm": "确认",
-    "compute_services": "计算服务",
```
Comment (PengJiyuan) [inline locale/lang.json]: ...

File: modules/network/pop/create_network/index.js [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+        flatState = refs.physical_network.state,
+        testFlat = /^\w+$/;
+      let phyiscalNet = refs.select_physical_network.state.value;
+      let vlanLength = phyNet.length;
+      let vlanNumber = parseInt(vlanState.value, 10);
+
+      let enableType = refs.enable_vlan && refs.enable_vlan.refs.enable_type.state.selectedValue;
+      let vlanItem = [];
+      let testMin, testMax;
+      phyNet.forEach(item => {
```
Comment (PengJiyuan) [inline modules/network/pop/create_network/index.js:200]: phyNet.forEach is not a function。
好好检查自己的代码，好好测试。

Comment (zkn111) [reply modules/network/pop/create_network/index.js:200]: 这个是因为你的配置文件没有neutron_network_vlanranges这个字段吧

File: modules/subnet/pop/create_subnet/add_hostroutes.jsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+let React = require('react');
+let {Button} = require('client/uskin/index');
+let __ = require('locale/client/admin.lang.json');
+
+class AddHostRoutes extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
Comment (PengJiyuan) [inline modules/subnet/pop/create_subnet/add_hostroutes.jsx:11]: // deep clone

**Review by PengJiyuan — CHANGES_REQUESTED** [review]

**Review by PengJiyuan — APPROVED** [review]

**Pull request merged**